src/t8_schemes/t8_quads_w_rectangular_subelements/t8_subelements/t8_subelements_quad_cxx.cxx

```diff
@@ -389,18 +415,38 @@ t8_subelement_scheme_quad_c::t8_element_is_family (t8_element_t ** fam)
 
 #ifdef T8_ENABLE_DEBUG
   int                 i;
-  for (i = 0; i < P4EST_CHILDREN; i++) {
-    /* at the moment, this function is only implemented for standard quad elements */
-    T8_ASSERT (pquad_w_sub_family[i]->dummy_is_subelement ==
-               T8_SUB_QUAD_IS_NO_SUBELEMENT);
 
+  /* TODO: this loop goes from 0 to 3 but with subelements there can be more elements in fam */
+  for (i = 0; i < P4EST_CHILDREN; i++) {
     T8_ASSERT (t8_element_is_valid (fam[i]));
   }
 #endif
-  return p4est_quadrant_is_family (&pquad_w_sub_family[0]->p4q,
-                                   &pquad_w_sub_family[1]->p4q,
-                                   &pquad_w_sub_family[2]->p4q,
-                                   &pquad_w_sub_family[3]->p4q);
+
+  /* Subelements can not be refined into other elements of a higher level. 
+   * So, if the first element of fam is a subelement, we assume that the following num_siblings 
+   * many elements are its siblings and therefore form a family. */
+  if (pquad_w_sub_family[0]->dummy_is_subelement == T8_SUB_QUAD_IS_SUBELEMENT) {
+    return 1;
+  }
```

Comment:
Yes you are right this test is very simplistic. I will think of a general solution. 
Is there a way to get the length of fam? For the 2D quad scheme we expect fam to consist of 4 elements but with subelements it could be more.

Reply:
This is the number ob siblings. `t8_element_num_siblings (element)`

Reply:
But I mean the number of elements fam points to. If I use t8_element_num_siblings(fam[0]) I get the number of siblings of the first element but this does not have to be the actual given number of elements in ** fam?

Reply:
Consider ** fam pointing to six elements and the first five elements do actually form a family of subelements. We would get `t8_element_num_siblings(fam[0]) = 5` and the following elements would give the same result and we would come to the conclusion that they form a family which is wrong because of the sixth element in fam.